cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaMirrorMaker2Cluster.java

```diff
@@ -0,0 +1,316 @@
+    protected static final String ENV_VAR_KAFKA_MIRRORMAKER_2_OAUTH_REFRESH_TOKENS_CLUSTERS = "KAFKA_MIRRORMAKER_2_OAUTH_REFRESH_TOKENS_CLUSTERS";
+
+    protected static final String OAUTH_SECRETS_BASE_VOLUME_MOUNT = "/opt/kafka/oauth/";
+
+    private List<KafkaMirrorMaker2ClusterSpec> clusters;
+
+    /**
+     * Constructor
+     *
+     * @param namespace Kubernetes/OpenShift namespace where Kafka Connect cluster
+     *                  resources are going to be created
+     * @param cluster   overall cluster name
+     */
+    private KafkaMirrorMaker2Cluster(String namespace, String cluster, Labels labels) {
+        super(namespace, cluster, labels);
+        this.name = KafkaMirrorMaker2Resources.deploymentName(cluster);
+        this.serviceName = KafkaMirrorMaker2Resources.serviceName(cluster);
+        this.ancillaryConfigName = KafkaMirrorMaker2Resources.metricsAndLogConfigMapName(cluster);
+    }
+
+    /**
+     * Creates instance of KafkaMirrorMaker2Cluster from CRD definition.
+     *
+     * @param kafkaMirrorMaker2 The Custom Resource based on which the cluster model should be created.
+     * @param versions The image versions for MirrorMaker 2.0 clusters.
+     * @return The MirrorMaker 2.0 cluster model.
+     */
+    public static KafkaMirrorMaker2Cluster fromCrd(KafkaMirrorMaker2 kafkaMirrorMaker2, 
+                                                   KafkaVersion.Lookup versions) {
+        KafkaMirrorMaker2Cluster cluster = new KafkaMirrorMaker2Cluster(kafkaMirrorMaker2.getMetadata().getNamespace(),
+                kafkaMirrorMaker2.getMetadata().getName(),
+                Labels.fromResource(kafkaMirrorMaker2).withKind(kafkaMirrorMaker2.getKind()));
+        KafkaMirrorMaker2Spec spec = kafkaMirrorMaker2.getSpec();
+        cluster.setOwnerReference(kafkaMirrorMaker2);
+        cluster.setImage(versions.kafkaMirrorMaker2Version(spec.getImage(), spec.getVersion()));
+        cluster.setConfiguration(new KafkaMirrorMaker2Configuration(spec.getConfig().entrySet())); 
+
+        List<KafkaMirrorMaker2ClusterSpec> clustersList = Optional.ofNullable(spec.getClusters())
+                .orElse(Collections.emptyList());
+        cluster.setClusters(clustersList);
+
+        String connectClusterAlias = spec.getConnectCluster();
+        if (connectClusterAlias != null) {
+            KafkaMirrorMaker2ClusterSpec connectCluster = clustersList.stream()
+                    .filter(clustersListItem -> connectClusterAlias.equals(clustersListItem.getAlias()))
+                    .findFirst()
+                    .orElseThrow(() -> new InvalidResourceException("connectCluster with alias " + connectClusterAlias + " cannot be found in the list of clusters at spec.clusters"));     
+
+            spec.setBootstrapServers(connectCluster.getBootstrapServers());
+            spec.setAuthentication(connectCluster.getAuthentication());
```

Comment:
Normally we treat the CR as read-only. I guess you're mutating it so you can call `fromSpec()` with it, right?

It might be better to make a copy of the CR, using the builder (and the `edit*()` methods it has) and pass that to `fromSpec()`. That would avoid the calling of this method having the surprising side-effect of changing the passed CR.

Reply:
Done